src/jarohen/with_open.clj

```diff
@@ -1,12 +1,50 @@
 (ns jarohen.with-open
-  (:import [java.io Closeable]
-           [java.lang AutoCloseable]))
+  (:import
+   [java.io Closeable]
+   [java.lang AutoCloseable]))
+
+(set! *warn-on-reflection* true)
+
+(defmacro try-finally
+  "Use try/finally such that an exception in the finally clause does not
```

Comment:
#suggestion Update the comment here to indicate the second arg is a marker like in `cond` and suggest the use of `:finally`.